backend/src/ee/services/dynamic-secret/providers/models.ts

```diff
@@ -7,6 +7,52 @@ export enum SqlProviders {
   MsSQL = "mssql"
 }
 
+export enum RedisProviders {
+  Redis = "redis",
+  Elasticache = "elasticache"
+}
+
+export const DynamicSecretRedisDBSchema = z
+  .object({
+    client: z.nativeEnum(RedisProviders),
+    host: z.string().trim().toLowerCase(),
+    port: z.number(),
+    username: z.string().trim(), // this is often "default".
+    password: z.string().trim().optional(),
+
+    elastiCacheIamUsername: z.string().trim().optional(),
```

Comment:
Next issue for me here is schema, we need a precise schema. 

That is as client changes the schema should also change using `discriminatedUnion` because the validation in refine works but won't be shown to end user. This is the issue we are facing with `integration` api at the moment.

The docs won't show this data in the json schema as its missing. We need the schema to show it. That is its redis only these values are needed